src/cromshell/status/command.py

```diff
@@ -114,34 +115,74 @@ def main(config, workflow_id):
     return ret_val
 
 
-def get_metadata_status_summary(workflow_metadata):
+def get_all_call_values(nested_dictionary):
+    """Summarizes a given workflow metadata dictionary into a
+    new dictionary with only task name and status"""
+    temp_dic1 = {}
+    temp_dic2 = {}
+    temp_dic3 = {}
+    for key, value in nested_dictionary.items():
+        # Skip any keys that have values that are strings, because we are note
+        # interested in them
+        if type(value) is str:
+            continue
+        # If a dictionary value is encountered then we iterate through the dictionary
+        # to find call statuses
+        if type(value) is dict:
+            temp_dic1 = get_all_call_values(value)
+        # If a list value is encountered then those are the calls and their status,
+        # we'll want to add the key and value to the temp dictionary
+        if type(value) is list:
+            # Each item in the list is the shard of a call so we'll itterate through
+            # the list and collect info on all shards
+            temp_dic3[key] = []
+            for i, shard in enumerate(value):
+                calls_status = shard
+                # If a key in the shard dictionary has the term "subWorkflowMetadata"
+                # then this is a subworkflow so we iterate through the dictionary to
+                # find call statuses
+                if "subWorkflowMetadata" in calls_status.keys():
+                    temp_dic2 = get_all_call_values(calls_status["subWorkflowMetadata"])
+                    # Remove "Scatter" key from dict
+                    if key in temp_dic3.keys():
```

Comment:
I don't quite understand the logic here. We can talk "in person" to go over it.

Reply:
In line 138 dictionary_3 is created using the keys from the dictionary being iterated through (example task1, task2, task3), however these can either be actual tasks or place holders for tasks from subworkflows (example ScatterAt98).  In line 144 we check if any of the associated tasks have subworkflows, if they do then we get their tasks and place into a dictionary_2 then remove the placeholder task from dictionary_3 because its not an actual task but a placeholder for subworkflow tasks.